File: Easy/Rotate by 90 degree/rotate-by-90-degree.py
#User function Template for python3



class Solution: 
    def rotateby90(self, a, n): 
        # Transpose
        for i in range(n):
            for j in range(i + 1, n):
                a[j][i], a[i][j] = a[i][j], a[j][i]
        for i in range(n // 2):  # Use // for integer division
            for j in range(n):
                a[n-1-i][j], a[i][j] = a[i][j], a[n-1-i][j]
        return a

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        n = int(input())
        matrix = [[0 for j in range(n)] for i in range(n)]
        line1 = [int(x) for x in input().strip().split()]
        k=0
        for i in range(n):
            for j in range (n):
                matrix[i][j]=line1[k]
                k+=1
        obj = Solution()
        obj.rotateby90(matrix,n)
        for i in range(n):
            for j in range(n):
                print(matrix[i][j],end=" ")
        print()

# } Driver Code Ends

File: Easy/Rotate by 90 degree/rotate-by-90-degree.cpp
//{ Driver Code Starts
#include <bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends


// class Solution: 
//     def rotateby90(self, a, n): 
//         # Transpose
//         for i in range(n):
//             for j in range(i + 1, n):
//                 a[j][i], a[i][j] = a[i][j], a[j][i]
//         for i in range(n // 2): # change below row with upper
//             for j in range(n):
//                 a[n-1-i][j], a[i][j] = a[i][j], a[n-1-i][j]
        // return a
class Solution{   
    public:
    //Function to rotate matrix anticlockwise by 90 degrees.
    void rotateby90(vector<vector<int> >& matrix, int n) {
        //1.take transpose
        for(int i=0;i<n;i++){
            for(int j=0;j<i;j++){
                swap(matrix[i][j],matrix[j][i]);
            }
        }
        //2.swap upper and lower rows;
        for(int i=0;i<(n/2);i++){
            for(int j=0;j<n;j++){
                swap(matrix[i][j],matrix[n-i-1][j]);
            }
        }
        
    } 
};


//{ Driver Code Starts.
int main() {
    int t;
    cin>>t;
    
    while(t--) 
    {
        int n;
        cin>>n;
        vector<vector<int> > matrix(n); 

        for(int i=0; i<n; i++)
        {
            matrix[i].assign(n, 0);
            for( int j=0; j<n; j++)
            {
                cin>>matrix[i][j];
            }
        }

        Solution ob;
        ob.rotateby90(matrix, n);
        for (int i = 0; i < n; ++i)
            for (int j = 0; j < n; ++j)
                cout<<matrix[i][j]<<" ";
        cout<<endl;
    }
    return 0;
}
// } Driver Code Ends